selfdrive/locationd/test/test_locationd.py

```diff
@@ -127,43 +130,53 @@ def send_msg(self, msg):
       waits_left -= 1
     time.sleep(0.0001)
 
-  def test_params_gps(self):
-    # first reset params
-    Params().put('LastGPSPosition', json.dumps({"latitude": 0.0, "longitude": 0.0, "altitude": 0.0}))
-
-    lat = 30 + (random.random() * 10.0)
-    lon = -70 + (random.random() * 10.0)
-    alt = 5 + (random.random() * 10.0)
-
-    for _ in range(1000):  # because of kalman filter, send often
-      msg = messaging.new_message('gpsLocationExternal')
-      msg.logMonoTime = 0
+  def get_fake_msg(self, name, t):
+    if name == "gpsLocationExternal":
+      msg = messaging.new_message(name)
       msg.gpsLocationExternal.flags = 1
       msg.gpsLocationExternal.verticalAccuracy = 1.0
       msg.gpsLocationExternal.speedAccuracy = 1.0
       msg.gpsLocationExternal.bearingAccuracyDeg = 1.0
       msg.gpsLocationExternal.vNED = [0.0, 0.0, 0.0]
-      msg.gpsLocationExternal.latitude = lat
-      msg.gpsLocationExternal.longitude = lon
-      msg.gpsLocationExternal.altitude = alt
-      self.send_msg(msg)
-
-    for _ in range(250):  # params is only written so often
-      msg = messaging.new_message('cameraOdometry')
-      msg.logMonoTime = 0
+      msg.gpsLocationExternal.latitude = self.lat
+      msg.gpsLocationExternal.longitude = self.lon
+      msg.gpsLocationExternal.altitude = self.alt
+    elif name == 'cameraOdometry':
+      msg = messaging.new_message(name)
       msg.cameraOdometry.rot = [0.0, 0.0, 0.0]
       msg.cameraOdometry.rotStd = [0.0, 0.0, 0.0]
       msg.cameraOdometry.trans = [0.0, 0.0, 0.0]
       msg.cameraOdometry.transStd = [0.0, 0.0, 0.0]
-      self.send_msg(msg)
+    elif name == 'sensorEvents':
```

Comment:
I would do something like this at the top:
```
try:
  msg = messaging.new_message(name)
except capnp.lib.capnp.KjException:
  msg = messaging.new_message(name, 0)

if name == "gpsLocationExternal":
  set stuff...
```

```suggestion
    elif name == 'sensorEvents':
```